packages/opentelemetry-resources/test/detect-resources.test.ts

```diff
@@ -28,7 +28,10 @@ describe('detectResourcesSync', () => {
   it('handles resource detectors which return Promise<Resource>', async () => {
     const detector: Detector = {
       async detect() {
-        return new Resource({ sync: 'fromsync' });
+        return new Resource(
+          { sync: 'fromsync' },
+          Promise.resolve({ async: 'fromasync' }).then(attrs => attrs)
```

Comment:
nit:

```suggestion
          Promise.resolve({ async: 'fromasync' })
```

Reply:
`Promise.resolve({ async: 'fromasync' })` returns a fulfilled promise right away, I'm changing it to `Promise.resolve().then(() => ({ async: 'fromasync' }))` to get the promise in the pending state, to reflect the actual async behavior of `HostDetector`